### PR TITLE
EE-8204 Adding BDD for hyphenated and apostrophe names

### PR DESCRIPTION
**Pull request by vbadiniHO**
Updating wiremock version to be able to use newer mock functionalities. Updated mock to match on name/surname initials, in addition to full json object. Added BDD for hyphenated/apostrophe names

### CODE REVIEW

File: src/test/groovy/acceptance/steps/NameMatchingSteps.groovy [hunk 1]
```diff
@@ -109,11 +109,33 @@ class NameMatchingSteps {
...
+            int nameLength = individual.getFirstName().length()
+            int surnameLength = individual.getLastName().length()
+            int nameIndex = 3
+            int surnameIndex = 3
+            if (nameLength < 3) {
+                nameIndex = nameLength
+            }
+            if (surnameLength < 3) {
+                surnameIndex = surnameLength
+            }
```
Comment (mikehalmaho) [inline src/test/groovy/acceptance/steps/NameMatchingSteps.groovy]: You could tidy this up using Math.min.
```
int nameIndex = min(3, individual.getFirstName().length())
```

File: src/test/groovy/acceptance/steps/NameMatchingSteps.groovy [hunk 2]
```diff
@@ -109,11 +109,33 @@ class NameMatchingSteps {
...
+
+            // Match json with initials of name and surname only
+            HMRC_MOCK_SERVICE.stubFor(post(urlEqualTo(INDIVIDUAL_MATCHING_ENDPOINT))
+                    .atPriority(1)
+                    .withRequestBody(matchingJsonPath("\$.firstName", matching(individual.getFirstName().substring(0, nameIndex)  + ".*")))
```
Comment (mikehalmaho) [inline src/test/groovy/acceptance/steps/NameMatchingSteps.groovy]: I thought that HMRC only check the first digit of the first name and 3 from the surname?

File: src/test/groovy/acceptance/steps/NameMatchingSteps.groovy (hunk 1 above)
Comment (vbadiniHO) [reply src/test/groovy/acceptance/steps/NameMatchingSteps.groovy]: +1

File: src/test/groovy/acceptance/steps/NameMatchingSteps.groovy (hunk 2 above)
Comment (vbadiniHO) [reply src/test/groovy/acceptance/steps/NameMatchingSteps.groovy]: Yes, I can change that :)

**Pull request merged**